lms/resources/lti_launch.py

```diff
@@ -42,18 +42,21 @@ def get_or_create_course(self):
         )
 
         legacy_course = course_service.get_or_create(authority_provided_id)
+        new_legacy_course = legacy_course in self._request.db.new
+
         course = course_service.upsert(
             authority_provided_id,
             context_id,
             params["context_title"],
             self._course_extra(),
             legacy_course.settings,
         )
+        new_course = course.id is None
 
-        if not self._request.db.is_modified(legacy_course) and not course.id:
+        if not new_legacy_course and new_course:
             LOG.warning(
-                "Created course from existing legacy course. consumer_key: %s",
-                legacy_course.consumer_key,
+                "Created course from existing legacy course. context_id: %s",
+                context_id,
```

Comment:
Logging the consumer_key was not the interesting bit. The course id is more interesting if we need to manually check something about it later.